source/MaterialXCore/Document.cpp

```diff
@@ -97,6 +97,18 @@ class Document::Cache
                         portElementMap.emplace(portElem->getQualifiedName(nodeName), portElem);
                     }
                 }
+                else
```

Comment:
Includes `nodegraph` references so can check if downstream ports reference a nodegraph.

Reply:
Note that this is outside the scope of this change but there is a lot of update here and could be prune by not updating for std lib nodes / graphs which should be "static" once loaded in. e.g. There are about 200 odd graphs to cache here on any change to the document.